nnvm/tests/python/compiler/test_top_level4.py

```diff
@@ -114,6 +116,22 @@ def test_reduce():
     verify_reduce((4, 4, 3), np.sum, sym.sum, axis=(0, 2))
     verify_reduce((4, 4, 3), np.sum, sym.sum)
 
+    def _with_keepdims(f):
+        """ Wrapper around numpy's argmax/argmin with `keepdims` argument supported """
+        def wrapper(x, axis=None, keepdims=False):
+            a = f(x, axis=axis)
+            if keepdims:
+                shape = list(a.shape)
+                shape.insert(axis,1)
+                a = a.reshape(tuple(shape))
+            return a
+        return wrapper
+
+    verify_reduce((4, 4, 3), _with_keepdims(np.argmax), sym.argmax, otype='int32')
+    verify_reduce((4, 4, 3), _with_keepdims(np.argmax), sym.argmax, oshape=(4,1,3), otype='int32', axis=1, keepdims=True)
+    verify_reduce((4, 4, 3), _with_keepdims(np.argmin), sym.argmin, otype='int32')
+    verify_reduce((4, 4, 3), _with_keepdims(np.argmin), sym.argmin, oshape=(4,1,3), otype='int32', axis=1, keepdims=True)
```

Comment:
Could you add a pattern which  keepdims is True and axis is None?

Reply:
Indeed, `axis=None` doesn't work, but the wrong shape is not the only reason. Looking into sources I would say that this value of axis is forbidden for every reduce operation in NNVM. Axis is defined as `struct ReduceParams { .. TShape axis; .. }`, but `FieldEntryBase::Set` doesn't have specializations for TShape and the default version rejects Nones with 

```
NNVMError: Invalid Parameter format for axis expect  but value='None'
```
(note also missing _type name)

I would prefer to not include None usecases in this PR, but solve this issue separately.

Reply:
Ok, I got it.